### DISCUSSION TITLE
Docs: Auto-generated og:images for all pages + SEO improvements

### DISCUSSION THREAD

**Discussion by loteoo** [directus/directus#11137] [Feature Requests]
While talking with @benhaynes , we discussed adding "auto-generated" og:images to the docs that are more useful than just a generic picture, using a Directus-powered "API"-like [service](https://thumbsmith.com/) that I'm working on. 

I looked at the VuePress config, and this can be easily done using the `extendPageData` hook that let's you modify meta tags on a page-per-page basis, programatically.

To push things forward, I started working on a PR a while back, but wanted to see your thoughts @rijkvanzanten before opening it.

Here's the diff of the potential PR:
https://github.com/directus/directus/compare/main...loteoo:feature/better-og-images

The docs pages have no sharing preview at the moment:

<img width="420" src="https://user-images.githubusercontent.com/14101189/149972965-c831ad10-162c-4d6a-b157-1d41cc7f8c6c.png" />

Here's what it could look like:

<img width="420" src="https://user-images.githubusercontent.com/14101189/149973400-272fa003-e4f8-4f13-b85e-c3ef9cfbd808.png" />

<details><summary><strong>More examples (click to expand):</strong></summary>

<img width="420" src="https://user-images.githubusercontent.com/14101189/149974338-01f2ea7e-98a7-47b2-8d88-fdcf14b72f1a.png" />

<img width="420" src="https://user-images.githubusercontent.com/14101189/149974473-12c1dc2a-0c93-41f5-a519-0befe803a895.png" />

Or with the new colors:

<img width="420" src="https://user-images.githubusercontent.com/14101189/149974615-2779e2df-f8e1-4888-8078-c249fd30faf4.png" />

<img width="420" src="https://user-images.githubusercontent.com/14101189/149974664-585def4f-dbf7-4b6e-bb06-b100cca9cbe1.png" />

</details>

#### Notes:
- Breadcrumb is built from the menu items
- Title and lastUpdated date comes from vuepress
- Read time comes from the `reading-time` package

#### Discussion points (either here in the discussion, or on the PR itself if this is greenlit):
- The design of the template
- A short page description could be added (in the picture, but also in the meta tags which is good for SEO), but requires adding a [vuepress exerpt](https://vuepress.vuejs.org/theme/writing-a-theme.html#content-excerpt) on all pages
- Possible integration in CI/CD with the thumbsmith CLI `deploy` command

@rijkvanzanten , I'd like to get your thoughts on this please! :)

**Comment by rijkvanzanten** [comment]
Love it! (cc @erondpowell)

**Reply by benhaynes** [reply]
I agree, I think this will be a great improvement to the Docs, Website, and other properties! We're undergoing some branding updates, so let me offer a few alternatives for design structure for consideration:

![OG Image 1](https://user-images.githubusercontent.com/522079/150217045-2299a67e-8e68-4743-8783-086dc66bbfd5.png)

![OG Image 2](https://user-images.githubusercontent.com/522079/150217065-e086c94f-6992-4339-b461-63dd2f59b004.png)

Down to iterate more, but this is a good first start!

**Reply by loteoo** [reply]
@benhaynes That looks great! I Especially like the first one. Should I update the template to match it? I'll open the PR at the same time 🙌

**Reply by benhaynes** [reply]
Let's do it! I can provide the vector assets or Figma used here if that's helpful. 🚀

**Reply by erondpowell** [reply]
> @benhaynes That looks great! I Especially like the first one. Should I update the template to match it? I'll open the PR at the same time 🙌

Yeah love that first one too!
Sorry to sound silly, but one question. Where are these thumbnails going to show up and be used at?
I haven't seen a place on the site that uses thumbnails for the docs pages.

**Reply by joselcvarela** [reply]
When we share a page from `docs.directus.io` it appears like this on Discord
<img width="400" src="https://user-images.githubusercontent.com/14039341/150287401-d5509efb-0502-4b6d-a494-b5cdf2c8d97b.png" />


With those thumbnails it should appear like this
<img width="400" src="https://user-images.githubusercontent.com/14039341/150287986-78223c0b-f999-4d0a-8ba9-8dda7164e26f.png" />

The same applies on Facebook, Twitter, etc.

You can preview those here for `directus.io`
https://www.opengraph.xyz/url/https%3A%2F%2Fdirectus.io/

**Reply by erondpowell** [reply]
Yes, of course! lol

**Reply by loteoo** [reply]
> Let's do it! I can provide the vector assets or Figma used here if that's helpful. 🚀

Sure yes actually I'll need those to do it 🤓 . Thanks! I'll package everything in the PR :)

**Reply by loteoo** [reply]
PR opened with updated screenshots! 🚀 

https://github.com/directus/directus/pull/11186

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
